package/MDAnalysis/coordinates/chain.py

```diff
@@ -417,6 +420,25 @@ def _get_local_frame(self, k):
         f = k - self._start_frames[i]
         return i, f
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state['ts'] = self.ts.__deepcopy__()
+        index = self.ts.frame
+        state.pop('_ChainReader__chained_trajectories_iter', None)
+        for reader in state['readers'][:self.__active_reader_index + 1]:
+            reader.rewind()
```

Comment:
why the rewind?

Reply:
If not, it will wind up into following error during pickling:
```
>>> self[self.ts.frame]
IndexError: Index 1 exceeds length of trajectory (1).
```
because the timestep.frame is "wrong" for these used/being used readers.

Reply:
is this still necessary with the recent cahnges to the chain reader?

Reply:
Yes, the `ts` of individual reader still has to be reset. I think I tried to reset the ts during iteration (e.g. inside `_get_local_frame` when reader_index is changed, but somehow failed (I don't remember why but seems to be a dead end to me)

Reply:
Please add a comment explaining why we are doing the rewinding here. The next person looking at it in a year's time will probably just take it out unless there's a comment here explaining WHY it's here.